examples/seismic/tutorials/09_viscoelastic.ipynb

```diff
@@ -0,0 +1,338 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## **Viscoelastic wave equation implementation on a staggered grid**\n",
+    "\n",
+    "This is a first attempt at implemenenting the viscoelastic wave equation as described in:\n",
+    "\n",
+    "[1] Johan O. A. Roberston, *et.al.* (1994). \"Viscoelatic finite-difference modeling\" GEOPHYSICS, 59(9), 1444-1456.\n",
+    "\n",
+    "See also the FDELMODC implementation by Jan Thorbecke:\n",
+    "\n",
+    "[2] https://janth.home.xs4all.nl/Software/fdelmodcManual.pdf\n",
+    "\n",
+    "In the following example, a three dimensional toy problem will be introduced consiting of a single Ricker source located at (100, 50, 35) in a 200 x 100 x 100 km domain."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Required imports:\n",
+    "import numpy as np\n",
+    "import sympy as sp\n",
+    "\n",
+    "from devito import *\n",
+    "from examples.seismic.source import RickerSource, TimeAxis\n",
+    "from examples.seismic import ModelViscoelastic, plot_image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Domain size:\n",
+    "extent = (200., 100., 100.)\n",
+    "h = 1.0\n",
+    "shape = (int(extent[0]/h+1), int(extent[1]/h+1), int(extent[2]/h+1))\n",
+    "\n",
+    "# Model physical parameters:\n",
+    "vp = np.zeros(shape)\n",
+    "qp = np.zeros(shape)\n",
+    "vs = np.zeros(shape)\n",
+    "qs = np.zeros(shape)\n",
+    "rho = np.zeros(shape)\n",
+    "\n",
+    "# Set up three horizontally separated layers:\n",
+    "for i in range(shape[0]):\n",
```

Comment:
This is very intricated for not much, ie for vp 
`vp = 1.52 * np.ones(shape);vp[:,:,0.5*extent:] = 1.6;vp[:,:,0.5*extent+4h:] = 2.2; `

Reply:
Yeah.